lighting/models/product_application_attachment.py

```diff
@@ -30,9 +30,26 @@ class LightingProductApplicationAttachment(models.Model):
         required=True,
     )
     attachment_id = fields.Many2one(
-        comodel_name="ir.attachment", compute="_compute_ir_attachment", readonly=True
+        comodel_name="ir.attachment",
+        compute="_compute_ir_attachment",
+        search="_search_attachment_id",
+        readonly=True,
     )
 
+    def _search_attachment_id(self, operator, value):
+        attachment_ids = (
+            self.env["ir.attachment"]
+            .search(
+                [
+                    ("res_model", "=", self._name),
+                    ("res_field", "=", "datas"),
+                    ("id", operator, value),
+                ]
+            )
+            .mapped("res_id")
+        )
+        return [("id", operator, attachment_ids)]
```

Comment:
```suggestion
        return [("id", "in", attachment_ids)]
```
It's always "in" here, you control the domain